apps/glusterfs/allocator_simple.go

```diff
@@ -95,23 +94,56 @@ func getDeviceListFromDB(db wdb.RODB, clusterId,
 	if err != nil {
 		return nil, err
 	}
+	return ring, nil
+}
 
+func GetRebalancedDeviceList(ring *SimpleAllocatorRing, brickId string) (SimpleDevices, error) {
 	devicelist := ring.GetDeviceList(brickId)
 
 	return devicelist, nil
 }
 
+func filterOfflineDevices(ring *SimpleAllocatorRing, clusterId, brickId string) *SimpleAllocatorRing {
```

Comment:
To me, it feels like this logic ought to be part of the same logic that populates the ring, rather than a secondary step that updates the node when it is built.

IOW, every place we're testing node.isOnline as input to the ring we can add a check on this node health.